### PR TITLE
Add an Android test project

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Add an Android test project and implement the required changes in Gradle. See the commit messages for details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/603)

### CODE REVIEW

**Comment by mnonnenmacher** [comment]
This still contains a placeholder commit, waiting for: https://github.com/heremaps/oss-review-toolkit/pull/602

File: analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml [hunk 1]
```diff
@@ -64,6 +64,21 @@ projects:
         dependencies: []
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml]: Commit message nits:

- Drop "the" before "scopes" in the second line.
- Drop "in the library project" in the second-to-last line.

**Comment by mnonnenmacher** [comment]
@sschuberth  Thanks for the review. FYI, this PR requires changes in the Travis configuration which I will prepare in a separate PR. Would you agree that it is sufficient to run the Android tests on Travis?

File: analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/gradle-all-dependencies-expected-result-with-curations.yml]: Done.

**Comment by sschuberth** [comment]
> Would you agree that it is sufficient to run the Android tests on Travis?

Probably. However, if it's not too much work, I'd prefer to see them in AppVeyor, too.

**Comment by mnonnenmacher** [comment]
Ok, I will look into installing the Android SDK on AppVeyor in another PR.

**Pull request merged**